### DISCUSSION TITLE
在下一个大版本中修改所有「QQ」相关字段

### DISCUSSION THREAD

**Discussion by stdrc** [howmanybots/onebot#76] [想法 / 建议]
这需要在各实现开始兼容 OneBot v11、生态重新活跃（各 SDK 和框架积极更新）之后才能做，可以使标准更加通用化

**Comment by shigma** [comment]
如果要兼容全平台会有一个问题，就是有些平台的用户 ID 是字符串（而不是数字），这个问题打算怎么解决呢？

**Comment by stdrc** [comment]
> 如果要兼容全平台会有一个问题，就是有些平台的用户 ID 是字符串（而不是数字），这个问题打算怎么解决呢？

那就直接改成字符串（

**Comment by chenxuan353** [comment]
> 如果要兼容全平台会有一个问题，就是有些平台的用户 ID 是字符串（而不是数字），这个问题打算怎么解决呢？

一般消息来源都会有唯一ID，就算没有也能生成，不然没办法原路径回复或者跨路径推送
可以参考我的兼容思路，我在python上写了一套[理论兼容框架](https://github.com/chenxuan353/tweetToBot/tree/v3)。
(技术有限，通用兼容层还是有问题，轻喷...)
不过也是还在完善中，生态不活跃也不好开发。。。

**Comment by undefined-moe** [comment]
我也建议全部换成字符串

**Comment by Ink-33** [comment]
@Shigma 写个断言，不是字符串就改成字符串（

**Comment by wxh06** [comment]
建议参考 Telegram 的 Bot 接口：https://core.telegram.org/bots
虽然 tg 的 bot 含有 QQ 原生不具备的实时交互功能（例如 H5 游戏等），但其 webhook 调用方法仍具参考价值

**Comment by shigma** [comment]
## CQ:at

目前的协议是 `[CQ:at,qq=xxx]`，考虑到跨平台，建议这样修改：


| 参数名 | 收 | 发 | 说明 |
| -- | -- | -- | -- |
| id | ✓ | ✓ | 平台 ID |
| role | ✓ | ✓ | 角色 ID |
| name | ✓ | ✓ | 显示的文本 |

特别地，`id` 可以取为 all 和 here，分别表示全体成员和在线成员。
